Engine/Annotations/Annotation.cs

```diff
@@ -1667,17 +1667,23 @@ public string Value
             {
                 get
                 {
-                    ITestStep value;
+                    object value;
                     if (member)
                     {
-                        value = (ITestStep) annotation.ParentAnnotation.Get<IObjectValueAnnotation>().Value;
+                        value = annotation.ParentAnnotation.Get<IObjectValueAnnotation>().Value;
                     }
                     else
                     {
-                        value = (ITestStep)annotation.Get<IObjectValueAnnotation>().Value;
+                        value = annotation.Get<IObjectValueAnnotation>().Value;
                     }
 
-                    return value.GetFormattedName();
+                    if (value is ITestStep step)
+                    {
+                        return step.GetFormattedName();
+                    }
+                    IEnumerable<ITestStep> steps = (IEnumerable<ITestStep>)value;
```

Comment:
Can we type match here instead of casting? And do something a bit more defensive than `steps.First()`